File: demo-web-java/src/main/java/com/example/demo/repository/UserRepositoryImpl.java
package com.example.demo.repository;

import com.example.demo.model.QUser;
import com.example.demo.model.User;
import com.querydsl.jpa.impl.JPAQuery;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import java.util.List;

public class UserRepositoryImpl implements UserRepositoryCustom {

  @PersistenceContext
  private EntityManager entityManager;

  @Override
  public User findByEmail(String email) {
    QUser user = QUser.user;

    JPAQuery<User> query = new JPAQuery<>(entityManager);
    query.from(user).where(user.email.eq(email));

    return query.fetchOne();
  }

  @Override
  public List<User> findActiveUsers() {
    QUser user = QUser.user;
    JPAQuery<User> query = new JPAQuery<>(entityManager);
    return query.from(user).select(QUser.create(user.id, user.username, user.email)).fetch();
  }
}

File: demo-web-scala/src/main/scala/com/example/demo/repository/UserRepositoryImpl.scala
package com.example.demo.repository

import java.util
import javax.persistence.{EntityManager, PersistenceContext}

import com.example.demo.model.{QUser, User, UsernameEmail}
import com.querydsl.jpa.impl.JPAQuery
import org.slf4j.LoggerFactory

class UserRepositoryImpl extends UserRepositoryCustom {

  var log = LoggerFactory.getLogger(classOf[UserRepositoryImpl])

  @PersistenceContext private val entityManager: EntityManager = null

  override def findActiveUsers(): util.ArrayList[UsernameEmail] = {
    def query = new JPAQuery[User](entityManager)

    val user = QUser as "user"
    query.from(user)
    val list = query.from(user).select(user.id, user.username, user.email).fetch()
    val result = new util.ArrayList[UsernameEmail];
    list.forEach(data => result.add(UsernameEmail(data.get(user.id), data.get(user.username), data.get(user.email))))
    result
  }

}
